hack/RELEASE.md

```diff
@@ -0,0 +1,77 @@
+(FIXME: Should this file live elsewhere?)
+
+# Libpod/Podman/Podman-Remote Automated Release Workflow
+
+1. Open a new PR.  The the `HEAD` commit *must* have a summary line matching the
+   following regex: `Release (v\d+\.\d+(\.\d+(-[\w\-\.]+)?)?)`.
+
+   * The second line of the commit text (after the summary and blank line), *must*
+     match the following regex: `^\(prior release: (.+)\)`.  Where the content of
```

Comment:
Do we need this? I would think we can programmatically determine this, given we're strict semantic-versioning now.

Reply:
My concern was over properly handling `X1.Y1.Z1-123-g0987654` upgrades to `X1.Y2`, but maybe you have a different understanding?  I only invented this (Prior release) method since it seemed a simple solution.

This is the failure-mode I was considering

1. Version `1.0.0` is tagged, branched, and released.
2. Somebody finds and fixes an (untested) `1.0` -> `2.0` upgrade breakage.
3. Version `1.0.1` is created on `1.0` branch (but not tagged or released)
4. An update is released for packages pointing at `1.0.1`
4. Other changes are made
5. Version `2.0.0` is submitted for automated release testing.
6. Automation (and humans) have established, successful upgrades MUST happen starting with `1.0.1` and not `1.0`.

In other words, how does a ***potential*** future release version know what is the ***minimum*** (lowest) acceptable prior version, for which upgrades (and downgrades?) are supported.

Since we cannot retroactively change any already released tag-objects, having this data available now will assist future automated release testing.  I thought that sticking something into the text is the simplest solution which is also, itself, easily verifiable.  If you know a better way, please tell (I'm ignorant in some things git and most things RCM :smile: ).

Or...maybe this is simply an unnecessary complication and I can just put it out of my mind.  I'm okay with that as well :grin: 

@edsantiago knows a thing or three about git...maybe he has an opinion (including: *"forget about upgrade-testing, it's too hard*" :smile: )

Reply:
Step 3 sounds dubious - we should always be tagging releases. Any version bump that is not -dev suffixed should make a tag.

Reply:
On upgrades in general - we're very lenient on this (RHEL means we have to be - we went from 1.0, to 1.4.2, to 1.6.4, to 1.9.3 - lots of skipping there). Given the way our internals work, at present it's doubtful we will need to implement a requirement that only major version N-1 be upgraded to new major version N.

Reply:
Thanks for the feedback/details, I think I'll just drop this aspect then.  However you brought up an interesting point: It sounds like there are at least two kinds of tags:  Those which pair to a release branch, and those that don't (e.g. -dev, -rc, etc.).

Idea: Maybe, for tag annotation then, we have automation add the branch-name of the release-branch.  That way other/different automation can easily tell the difference between release-tags and other-tags.

Or do you have another idea for the tag messages that might be helpful?